connector/mqtt/vernemq/v2k-brigde/app/AgentMessenger.js

```diff
@@ -0,0 +1,56 @@
+const { IoTAgent } = require('@dojot/iotagent-nodejs');
+const { logger } = require('@dojot/dojot-module-logger');
+const defaultConfig = require('./config');
+const Utils = require('./utils/utils');
+
+const TAG = { filename: 'agent-messenger' };
+
+class AgentMessenger {
+  constructor(config) {
+    this.initialized = false;
+    this.config = config || defaultConfig;
+
+    /* set log level */
+    logger.setLevel(this.config.app.mqtt_log_level);
+
+    this.iotagent = new IoTAgent();
+  }
+
+  init(mqttClient) {
+    this.iotagent.init().then(() => {
+      logger.debug('... IoT agent was initialized', TAG);
+
+      logger.debug('Registering callbacks for device events...', TAG);
+      this.iotagent.on('iotagent.device', 'device.create', (tenant) => {
+        logger.debug(`Got device creation message. Tenant is ${tenant}.`, TAG);
+        logger.debug('Got configure event from Device Manager', TAG);
+      });
+      logger.debug('... callbacks for device events were registered.', TAG);
+      logger.debug(`Subscribing to vernemq with topic: ${this.config.mqtt.subscribeTopic}`, TAG);
+      // subscribe to verne
+      mqttClient.subscribe(this.config.mqtt.subscribeTopic);
+    }).catch(() => {
+      logger.error('An error occurred while initializing the IoTAgent. Bailing out!', TAG);
```

Comment:
Which error? It Should be nice to print it.

Reply:
The iot-agent library, catch with no error message. As a generic error, it cannot describe for more details.